lib/spack/spack/test/ci.py

```diff
@@ -126,32 +141,39 @@ def test_register_cdash_build():
         ci.register_cdash_build(build_name, base_url, project, site, track)
 
 
-def test_relate_cdash_builds(config, mock_packages):
-    root_spec = (
-        'eJztkk1uwyAQhfc5BbuuYjWObSKuUlURYP5aDBjjBPv0RU7iRI6qpKuqUtnxzZvRwHud'
-        'YxSt1oCMyuVoBdI5MN8paxDYZK/ZbkLYU3kqAuA0Dtz6BgGtTB8XdG87BCgzwXbwXArY'
-        'CxYQiLtqXxUTpLZxSjN/mWlwwxAQlJ7v8wpFtsvK1UXSOUyTjvRKB2Um7LBPhZD0l1md'
-        'xJ7VCATfszOiXGOR9np7vwDn7lCMS8SXQNf3RCtyBTVzzNTMUMXmfWrFeR+UngEAEncS'
-        'ASjKwZcid7ERNldthBxjX46mMD2PsJnlYXDs2rye3l+vroOkJJ54SXgZPklLRQmx61sm'
-        'cgKNVFRO0qlpf2pojq1Ro7OG56MY+Bgc1PkIo/WkaT8OVcrDYuvZkJdtBl/+XCZ+NQBJ'
-        'oKg1h6X/VdXRoyE2OWeH6lCXZdHGrauUZAWFw/YJ/0/39OefN3F4Kle3cXjYsF684ZqG'
-        'Tbap/uPwbRx+YPStIQ8bvgA7G6YE'
-    )
+def test_relate_cdash_builds(config, mutable_mock_env_path, mock_packages):
+    e = ev.create('test1')
+    e.add('dyninst')
+    e.concretize()
+
+    dyninst_hash = None
+    hash_dict = {}
 
-    dep_builds = 'diffutils;libiconv'
-    spec_map = ci.get_concrete_specs(root_spec, 'bzip2', dep_builds, 'NONE')
-    cdash_api_url = 'http://cdash.fake.org'
-    job_build_id = '42'
-    cdash_project = 'spack'
-    cdashids_mirror_url = 'https://my.fake.mirror'
+    with e as active_env:
+        for s in active_env.all_specs():
+            hash_dict[s.name] = s.build_hash()
+            if s.name == 'dyninst':
+                dyninst_hash = s.build_hash()
 
-    with pytest.raises(web_util.SpackWebError):
-        ci.relate_cdash_builds(spec_map, cdash_api_url, job_build_id,
-                               cdash_project, cdashids_mirror_url)
+        assert(dyninst_hash)
 
-    # Just make sure passing None for build id doesn't throw exceptions
-    ci.relate_cdash_builds(spec_map, cdash_api_url, None, cdash_project,
-                           cdashids_mirror_url)
+        dep_builds = 'libdwarf;libelf'
+        spec_map = ci.get_concrete_specs(
+            active_env, dyninst_hash, 'dyninst', dep_builds, 'NONE')
+        assert('root' in spec_map and 'deps' in spec_map)
+
+        cdash_api_url = 'http://cdash.fake.org'
+        job_build_id = '42'
+        cdash_project = 'spack'
+        cdashids_mirror_url = 'https://my.fake.mirror'
+
+        with pytest.raises(spack.error.SpackError):
+            ci.relate_cdash_builds(spec_map, cdash_api_url, job_build_id,
+                                   cdash_project, [cdashids_mirror_url])
+
+        # Just make sure passing None for build id doesn't throw exceptions
+        ci.relate_cdash_builds(spec_map, cdash_api_url, None, cdash_project,
+                               [cdashids_mirror_url])
```

Comment:
Yeah, that was kind of useless.  I fixed this and a few other tests to mock `build_opener` from `six.moves.urllib.request` to return something I can use to actually check things in methods that make http requests.

Reply:
My comment was intended to indicate I think I've resolved this, as this test (and a few others).  I did leave in this one call that passes a build_id of `None` because there's a check for that at the top of the method, and I wanted that line covered.  But notice that test now makes two calls to the same method above that actually make useful assertions.